### PR TITLE
fix(analyzer): add missing warning in outcome

### LINKED ISSUE #1652: Warn isn't working with regex analyzer
**Bug Description**

I'm trying to use **Warn** as an outcome for regex analyzer but it doesn't work, it shows fail instead

**Expected Behavior**

Just like the examples here https://troubleshoot.sh/docs/analyze/ , i expected that warn is supported as an option.

**Steps To Reproduce**

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: SupportBundle
metadata:
  name: test
spec:
  collectors:
    - clusterResources:
        exclude: true
    - clusterInfo:
        exclude: true
  hostCollectors:
    - run:
        collectorName: test
        command: "sh"
        args:
          - -c
          - |
            #!/bin/bash
            echo "testing"

  analyzers:
    - textAnalyze:
        checkName:  "test"
        fileName: host-collectors/run-host/test.txt
        regex: 'Check passed'
        outcomes:
          - pass:
              when: "true"
              message: "Check passed."
          - warn:
              when: "false"
              message: "There seems to be an issue with the test."
```

**Additional Context**

Include the following information.
- Replicated Troubleshoot 0.85.0
- MacOS
- Sequoia 15.0.1

### PR DESCRIPTION
**Pull request by DexterYan**
## Description, Motivation and Context

- added missing outcome condition for warning
- added tests for both warning and fail

[sc-115535](https://app.shortcut.com/replicated/story/115535/regex-analyzer-warn-doesn-t-work)

## Checklist

- [x] New and existing tests pass locally with introduced changes.
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) are informative and highlight any breaking changes
- [ ] Any documentation required has been added/updated. For changes to https://troubleshoot.sh/ create a PR [here](https://github.com/replicatedhq/troubleshoot.sh/pulls)

## Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: pkg/analyze/text_analyze.go [hunk 1]
```diff
@@ -115,54 +115,66 @@ func analyzeRegexPattern(pattern string, collected []byte, outcomes []*troublesh
...
-	passWhen := true
-	if passOutcome != nil && passOutcome.When != "" {
-		passWhen, err = strconv.ParseBool(passOutcome.When)
-		if err != nil {
-			return nil, errors.Wrapf(err, "failed to process when statement: %s", passOutcome.When)
-		}
-	}
+	for _, outcome := range outcomes {
+		if outcome.Fail != nil {
+			isMatch := re.MatchString(string(collected))
```
Comment (nvanthao) [inline pkg/analyze/text_analyze.go]: this can be called only once before the loop

File: pkg/analyze/text_analyze.go [hunk 2]
```diff
@@ -115,54 +115,66 @@ func analyzeRegexPattern(pattern string, collected []byte, outcomes []*troublesh
...
-		return &result, nil
-	}
+			if isMatch == failWhen {
+				result.IsFail = true
+				result.IsWarn = false
+				result.Message = outcome.Fail.Message
+				result.URI = outcome.Fail.URI
+			}
+		} else if outcome.Warn != nil {
+			isMatch := re.MatchString(string(collected))
```
Comment (nvanthao) [inline pkg/analyze/text_analyze.go]: this can be called only once before the loop

File: pkg/analyze/text_analyze.go [hunk 3]
```diff
@@ -115,54 +115,66 @@ func analyzeRegexPattern(pattern string, collected []byte, outcomes []*troublesh
...
+				return nil, errors.Wrapf(err, "failed to process when statement: %s", outcome.Warn.When)
+			}
+
+			if isMatch == warnWhen {
+				result.IsWarn = true
+				result.Message = outcome.Warn.Message
+				result.URI = outcome.Warn.URI
+			}
+		} else if outcome.Pass != nil {
+			isMatch := re.MatchString(string(collected))
```
Comment (nvanthao) [inline pkg/analyze/text_analyze.go]: this can be called only once before the loop

**Review by nvanthao — CHANGES_REQUESTED** [review]
just a nit otherwise LGTM

File: pkg/analyze/text_analyze.go (hunk 1 above)
Comment (DexterYan) [reply pkg/analyze/text_analyze.go]: good catch! Fixed, Thx!

**Review by nvanthao — APPROVED** [review]
LGTM 👍

**Pull request merged**